### PR TITLE
STOR-596: Extract all k8s resources into static YAMLs

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/STOR-596
cc @openshift/storage 
/assign @gnufied 
/assign @bertinatto

### CODE REVIEW

File: assets/assets.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package assets
+
+import (
+	"embed"
+	"strings"
+)
+
+//go:embed *
```
Comment (sp98) [inline assets/assets.go:8]: won't this need golang 1.16?   I'm getting error while building the images. 

```assets/assets.go:4:2: cannot find package "." in:
	/go/src/github.com/openshift/local-storage-operator/vendor/embed
make: *** [Makefile:228: build-operator] Error 1```

**Review by sp98 — CHANGES_REQUESTED** [review]

**Review by sp98 — CHANGES_REQUESTED** [review]
Test a bit. Looks good so far. Need to test more.

Some nits.

File: assets/assets.go (hunk 1 above)
Comment (dobsonj) [reply assets/assets.go:8]: Yes, it requires 1.16. I check `go.mod` and it already has 1.16 listed:
https://github.com/openshift/local-storage-operator/blob/8feb36c44af009fae1fc9bf18b525bcbfc536e1d/go.mod#L3
Are there any other places that should be updated?

Comment (gnufied) [reply assets/assets.go:8]: I think we will have the Dockerfile

File: controllers/localvolume/localvolume_controller.go [hunk 2]
```diff
@@ -330,23 +335,20 @@ func addOwnerLabels(meta *metav1.ObjectMeta, cr *localv1.LocalVolume) bool {
...
+			"${OBJECT_NAME}", scName,
 		},
-		Provisioner:       "kubernetes.io/no-provisioner",
-		ReclaimPolicy:     &deleteReclaimPolicy,
-		VolumeBindingMode: &firstConsumerBinding,
+	)
+	if err != nil {
+		return nil, err
 	}
+	sc := resourceread.ReadStorageClassV1OrDie(scBytes)
```
[the remark this answers is not in the document]
Comment (gnufied) [reply controllers/localvolume/localvolume_controller.go:348]: I personally think it is fine to use common functions that help us in reading resources etc. As long as we do not switch to parallel controller loop style of library-go, we are probably fine.

[the remark this answers is not in the document]
Comment (dobsonj) [reply controllers/localvolume/localvolume_controller.go:348]: There was already a dependency on library-go for resourcemerge:
https://github.com/openshift/local-storage-operator/blob/5e3f542b1f29d2298548a75ee964f1ca5330288f/vendor/modules.txt#L154-L157
used here:
https://github.com/openshift/local-storage-operator/blob/5e3f542b1f29d2298548a75ee964f1ca5330288f/controllers/localvolume/resourceapply.go#L26
So this change just pulls in resourceread along with it.

File: assets/assets.go (hunk 1 above)
Comment (dobsonj) [reply assets/assets.go:8]: Good catch, updated.

File: assets/templates/diskmaker-discovery-daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          name: local-disks
+        - mountPath: /dev
+          mountPropagation: HostToContainer
+          name: device-dir
+        - mountPath: /run/udev
+          mountPropagation: HostToContainer
+          name: run-udev
+      hostPID: true
+      priorityClassName: openshift-user-critical
+      serviceAccount: local-storage-admin
```
[the remark this answers is not in the document]
Comment (dobsonj) [reply assets/templates/diskmaker-discovery-daemonset.yaml:55]: Thanks for pointing this out, it cleared up a point of confusion for me in daemonset_test.go. `serviceAccount` is mapped to `DeprecatedServiceAccount` here:
https://github.com/openshift/local-storage-operator/blob/5e3f542b1f29d2298548a75ee964f1ca5330288f/vendor/k8s.io/api/core/v1/types.go#L3010-L3018
So the fact that `serviceAccount` was defined in the yaml was causing `DeprecatedServiceAccount` to be set in the template during the unit test, but we don't actually copy that deprecated field in `MutateAggregatedSpec()`. I don't see anything else in LSO that references either of those, so I think it's safe to remove. Looks like `ServiceAccountName` is the only one needed.

**Review by sp98 — CHANGES_REQUESTED** [review]
Testing looks good. One small comment. Otherwise looks good to me.

File: controllers/nodedaemon/daemonsets.go [hunk 4]
```diff
@@ -116,90 +111,78 @@ func getLocalProvisionerDSMutateFn(
 ) func(*appsv1.DaemonSet) error {
 
 	return func(ds *appsv1.DaemonSet) error {
-		name := ProvisionerName
+		// read template for default values
```
[the remark this answers is not in the document]
Comment (rohantmp) [reply controllers/nodedaemon/daemonsets.go:114]: We can remove it!

**Comment by gnufied** [comment]
@sp98 did we had any outstanding concerns on this pr?

**Comment by gnufied** [comment]
lets remove `getLocalProvisionerDSMutateFn` in a follow up.

**Pull request merged**